e2e/tests/functional/plugins/notebook/notebookWithCouchDB.e2e.spec.js

```diff
@@ -0,0 +1,232 @@
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/*
+This test suite is dedicated to tests which verify the basic operations surrounding Notebooks with CouchDB.
+*/
+
+const { test, expect } = require('../../../../baseFixtures');
+const { createDomainObjectWithDefaults } = require('../../../../appActions');
+
+// Try to reduce indeterminism of browser requests by only returning fetch requests.
+// Filter out preflight CORS, fetching stylesheets, page icons, etc. that can occur during tests
+function filterNonFetchRequests(requests) {
+    return requests.filter(request => {
+        return (request.resourceType() === 'fetch');
+    });
+}
+
+test.describe('Notebook Network Request Inspection @couchdb', () => {
+    let testNotebook;
+    test.beforeEach(async ({ page }) => {
+        //Navigate to baseURL
+        await page.goto('./', { waitUntil: 'networkidle' });
+
+        // Create Notebook
+        testNotebook = await createDomainObjectWithDefaults(page, {
+            type: 'Notebook',
+            name: "TestNotebook"
+        });
+    });
+
+    test('Inspect Notebook Entry Network Requests', async ({ page }) => {
+        // Expand sidebar
+        await page.locator('.c-notebook__toggle-nav-button').click();
+
+        // Collect all request events to count and assert after notebook action
+        let requests = [];
+        page.on('request', (request) => requests.push(request));
+
+        let [notebookUrlRequest, allDocsRequest] = await Promise.all([
+            // Waits for the next request with the specified url
+            page.waitForRequest(`**/openmct/${testNotebook.uuid}`),
+            page.waitForRequest('**/openmct/_all_docs?include_docs=true'),
+            // Triggers the request
+            page.click('text=Page Add >> button'),
+            // Ensures that there are no other network requests
+            page.waitForLoadState('networkidle')
+        ]);
+        // Assert that only two requests are made
+        // Network Requests are:
+        // 1) The actual POST to create the page
+        // 2) The shared worker event from 👆 request
+        expect(requests.length).toBe(2);
+
+        // Assert on request object
+        expect(notebookUrlRequest.postDataJSON().metadata.name).toBe('TestNotebook');
+        expect(notebookUrlRequest.postDataJSON().model.persisted).toBeGreaterThanOrEqual(notebookUrlRequest.postDataJSON().model.modified);
+        expect(allDocsRequest.postDataJSON().keys).toContain(testNotebook.uuid);
+
+        // Add an entry
+        // Network Requests are:
+        // 1) The actual POST to create the entry
+        // 2) The shared worker event from 👆 POST request
+        requests = [];
```

Comment:
maybe we could make this requests var more meaningful? addNotebookEntryRequests or something like that